core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPreferWritePartitioning.java

```diff
@@ -0,0 +1,95 @@
+package io.trino.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.Session;
+import io.trino.sql.planner.Partitioning;
+import io.trino.sql.planner.PartitioningScheme;
+import io.trino.sql.planner.assertions.PlanMatchPattern;
+import io.trino.sql.planner.iterative.rule.test.RuleAssert;
+import io.trino.sql.planner.iterative.rule.test.RuleTester;
+import io.trino.sql.planner.plan.PlanNodeId;
+import io.trino.sql.planner.plan.ValuesNode;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
+import static io.trino.SystemSessionProperties.USE_PREFERRED_WRITE_PARTITIONING;
+import static io.trino.sql.planner.SystemPartitioningHandle.FIXED_HASH_DISTRIBUTION;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.tableWriter;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.planner.iterative.rule.test.RuleTester.defaultRuleTester;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public class TestPreferWritePartitioning
+{
+    private static final String MOCK_CATALOG = "mock_catalog";
+    private static final String TEST_SCHEMA = "test_schema";
+
+    private static final Session SESSION = testSessionBuilder()
+            .setCatalog(MOCK_CATALOG)
+            .setSchema(TEST_SCHEMA)
+            .build();
+    private static final Session SESSION_WITH_PREFERRED_PARTITIONING = testSessionBuilder()
+            .setCatalog(MOCK_CATALOG)
+            .setSchema(TEST_SCHEMA)
+            .setSystemProperty(USE_PREFERRED_WRITE_PARTITIONING, "true")
+            .build();
+
+    private static final PlanMatchPattern MATCH_PATTERN = tableWriter(
```

Comment:
it's used once. Inline it in test

Reply:
It is used twice. Second usage is in the second commit